sherpa/astro/plot.py

```diff
@@ -64,25 +64,49 @@ def _check_hist_bins(plot):
 
     Parameters
     ----------
-    plot
-        The plot structure, which must have xlo and xhi attributes.
+    xlo, xhi : array
+        Lower and upper bin boundaries. Typically, ``xlo`` will contain the
+        lower boundary and ``xhi`` the upper boundary, but this function can
+        deal with situations where that is reversed. Both arrays have to be
+        monotonically increasing or decreasing.
+
+    Returns
+    -------
+    xlo, xhi : array
+        xlo and xhi with values that were very close (within numerical
+        tolerance) before changed such that they now match exactly.
 
     Notes
     -----
     Note that this holds even when plotting wavelength values, who
     have xlo/xhi in decreasing order, since the lo/hi values still
     hold.
-
     """
+    if len(xlo) != len(xhi):
+        # Not a Sherpa specific error, because this is more for developers.
+        raise ValueError('Input arrays must have same length.')
+    # Nothing to compare if input arrays are empty.
+    if len(xlo) == 0:
+        return xlo, xhi
 
     # Technically idx should be 0 or 1, with no -1 values. We
     # do not enforce this. What we do is to take all bins that
     # appear similar (sao_fcmp==0) and set the xlo[i+1] bin
     # to the xhi[i] value.
     #
-    equal = sao_fcmp(plot.xlo[1:], plot.xhi[:-1], _tol)
+    # Deal with xhi <-> xlo switches. Those can occor when converting
```

Comment:
occor -> occur but I think the first two lines can get removed as it repeated in "Deal with reversed order." I think